File: 434. Number of Segments in a String/_434.py
class Solution:
    def countSegments(self, s: str) -> int:
        flag = False
        count = 0
        for i in s:
            if i != ' ' and not flag:
                count += 1
                flag = True
            if i == ' ':
                flag = False
        
        return count

    def countSegments1(self, s: str) -> int:
        return len(s.split())

File: 434. Number of Segments in a String/434.go
package leetcode

func countSegments(s string) int {
	flag := false
	count := 0
	for _, v := range s{
		if v!=' ' && !flag{
			count++
			flag = true
		}
		if v==' '{
			flag = false
		}
	}
	return count
}